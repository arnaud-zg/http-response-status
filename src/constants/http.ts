export namespace NHttpHeaders {
  export enum EHttpHeaders {
    ACCESS_CONTROL_ALLOW_CREDENTIALS = 'Access-Control-Allow-Credentials',
    ACCESS_CONTROL_ALLOW_HEADERS = 'Access-Control-Allow-Headers',
    ACCESS_CONTROL_ALLOW_METHODS = 'Access-Control-Allow-Methods',
    ACCESS_CONTROL_ALLOW_ORIGIN = 'Access-Control-Allow-Origin',
    ACCESS_CONTROL_EXPOSE_HEADERS = 'Access-Control-Expose-Headers',
    ACCESS_CONTROL_MAX_AGE = 'Access-Control-Max-Age',
    ACCEPT_RANGES = 'Accept-Ranges',
    AGE = 'Age',
    ALLOW = 'Allow',
    ALTERNATE_PROTOCOL = 'Alternate-Protocol',
    CACHE_CONTROL = 'Cache-Control',
    CLIENT_DATE = 'Client-Date',
    CLIENT_PEER = 'Client-Peer',
    CLIENT_RESPONSE_NUM = 'Client-Response-Num',
    CONNECTION = 'Connection',
    CONTENT_DISPOSITION = 'Content-Disposition',
    CONTENT_ENCODING = 'Content-Encoding',
    CONTENT_LANGUAGE = 'Content-Language',
    CONTENT_LENGTH = 'Content-Length',
    CONTENT_LOCATION = 'Content-Location',
    CONTENT_MD5 = 'Content-MD5',
    CONTENT_RANGE = 'Content-Range',
    CONTENT_SECURITY_POLICY = 'X-Content-Security-Policy',
    X_WEBKIT_CSP = 'Content-Security-Policy, X-Content-Security-Policy, X-Webkit-CSP',
    CONTENT_SECURITY_POLICY_REPORT_ONLY = 'Content-Security-Policy-Report-Only',
    CONTENT_TYPE = 'Content-Type',
    DATE = 'Date',
    ETAG = 'Etag',
    EXPIRES = 'Expires',
    HTTP = 'HTTP',
    KEEP_ALIVE = 'Keep-Alive',
    LAST_MODIFIED = 'Last-Modified',
    LINK = 'Link',
    LOCATION = 'Location',
    P3P = 'P3P',
    PRAGMA = 'Pragma',
    PROXY_AUTHENTICATE = 'Proxy-Authenticate',
    PROXY_CONNECTION = 'Proxy-Connection',
    REFRESH = 'Refresh',
    RETRY_AFTER = 'Retry-After',
    SERVER = 'Server',
    SET_COOKIE = 'Set-Cookie',
    STATUS = 'Status',
    STRICT_TRANSPORT_SECURITY = 'Strict-Transport-Security',
    TIMING_ALLOW_ORIGIN = 'Timing-Allow-Origin',
    TRAILER = 'Trailer',
    TRANSFER_ENCODING = 'Transfer-Encoding',
    UPGRADE = 'Upgrade',
    VARY = 'Vary',
    VIA = 'Via',
    WARNING = 'Warning',
    WWW_AUTHENTICATE = 'WWW-Authenticate',
    X_ASPNET_VERSION = 'X-Aspnet-Version',
    X_CONTENT_TYPE_OPTIONS = 'X-Content-Type-Options',
    X_FRAME_OPTIONS = 'X-Frame-Options',
    X_PERMITTED_CROSS_DOMAIN_POLICIES = 'X-Permitted-Cross-Domain-Policies',
    X_PINGBACK = 'X-Pingback',
    X_POWERED_BY = 'X-Powered-By',
    X_ROBOTS_TAG = 'X-Robots-Tag',
    X_UA_COMPATIBLE = 'X-UA-Compatible',
    X_XSS_PROTECTION = 'X-XSS-Protection',
  }
}

export namespace NHttpStatuses {
  export enum EInformational {
    CONTINUE = 100,
    SWITCHING_PROTOCOLS = 101,
    PROCESSING = 102,
  }

  export enum ESuccess {
    OK = 200,
    CREATED = 201,
    ACCEPTED = 202,
    NON_AUTHORITATIVE_INFORMATION = 203,
    NO_CONTENT = 204,
    RESET_CONTENT = 205,
    PARTIAL_CONTENT = 206,
    MULTI_STATUS = 207,
    ALREADY_REPORTED = 208,
    IM_USED = 226,
  }

  export enum ERedirection {
    MULTIPLE_CHOICES = 300,
    MOVED_PERMANENTLY = 301,
    FOUND = 302,
    SEE_OTHER = 303,
    NOT_MODIFIED = 304,
    USE_PROXY = 305,
    TEMPORARY_REDIRECT = 307,
    PERMANENT_REDIRECT = 308,
  }

  export enum ClientError {
    BAD_REQUEST = 400,
    UNAUTHORIZED = 401,
    PAYMENT_REQUIRED = 402,
    FORBIDDEN = 403,
    NOT_FOUND = 404,
    METHOD_NOT_ALLOWED = 405,
    NOT_ACCEPTABLE = 406,
    PROXY_AUTHENTICATION_REQUIRED = 407,
    REQUEST_TIMEOUT = 408,
    CONFLICT = 409,
    GONE = 410,
    LENGTH_REQUIRED = 411,
    PRECONDITION_FAILED = 412,
    PAYLOAD_TOO_LARGE = 413,
    REQUEST_URI_TOO_LONG = 414,
    UNSUPPORTED_MEDIA_TYPE = 415,
    REQUESTED_RANGE_NOT_SATISFIABLE = 416,
    EXPECTATION_FAILED = 417,
    I_M_A_TEAPOT = 418,
    MISDIRECTED_REQUEST = 421,
    UNPROCESSABLE_ENTITY = 422,
    LOCKED = 423,
    FAILED_DEPENDENCY = 424,
    UPGRADE_REQUIRED = 426,
    PRECONDITION_REQUIRED = 428,
    TOO_MANY_REQUESTS = 429,
    REQUEST_HEADER_FIELDS_TOO_LARGE = 431,
    CONNECTION_CLOSED_WITHOUT_RESPONSE = 444,
    UNAVAILABLE_FOR_LEGAL_REASONS = 451,
    CLIENT_CLOSED_REQUEST = 499,
  }

  export enum EServerError {
    INTERNAL_SERVER_ERROR = 500,
    NOT_IMPLEMENTED = 501,
    BAD_GATEWAY = 502,
    SERVICE_UNAVAILABLE = 503,
    GATEWAY_TIMEOUT = 504,
    HTTP_VERSION_NOT_SUPPORTED = 505,
    VARIANT_ALSO_NEGOTIATES = 506,
    INSUFFICIENT_STORAGE = 507,
    LOOP_DETECTED = 508,
    NOT_EXTENDED = 510,
    NETWORK_AUTHENTICATION_REQUIRED = 511,
    NETWORK_CONNECT_TIMEOUT_ERROR = 599,
  }
}
