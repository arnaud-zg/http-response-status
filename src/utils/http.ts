import { NHttpStatuses } from '../constants/http';

const HTTP_STATUS_CLIENT_ERROR = 'ClientError';
const HTTP_STATUS_INFORMATIONAL = 'Informational';
const HTTP_STATUS_REDIRECTION = 'Redirection';
const HTTP_STATUS_SERVER_ERROR = 'ServerError';
const HTTP_STATUS_SUCCESS = 'Success';

export type THttpStatusCodeCatagory = ReturnType<typeof getHttpStatusCatgory>;

export type THttpStatusName = ReturnType<typeof getHttpStatusName>;

export const getHttpStatusName = (httpStatusCode: number) =>
  NHttpStatuses.EInformational[httpStatusCode] ||
  NHttpStatuses.ESuccess[httpStatusCode] ||
  NHttpStatuses.ERedirection[httpStatusCode] ||
  NHttpStatuses.ClientError[httpStatusCode] ||
  NHttpStatuses.EServerError[httpStatusCode];

export const getHttpStatusCatgory = (httpStatusCode: number) => {
  if (
    Object.values(NHttpStatuses.EInformational).indexOf(httpStatusCode) !== -1
  ) {
    return HTTP_STATUS_INFORMATIONAL;
  } else if (
    Object.values(NHttpStatuses.ESuccess).indexOf(httpStatusCode) !== -1
  ) {
    return HTTP_STATUS_SUCCESS;
  } else if (
    Object.values(NHttpStatuses.ERedirection).indexOf(httpStatusCode) !== -1
  ) {
    return HTTP_STATUS_REDIRECTION;
  } else if (
    Object.values(NHttpStatuses.ClientError).indexOf(httpStatusCode) !== -1
  ) {
    return HTTP_STATUS_CLIENT_ERROR;
  } else if (
    Object.values(NHttpStatuses.EServerError).indexOf(httpStatusCode) !== -1
  ) {
    return HTTP_STATUS_SERVER_ERROR;
  }

  return undefined;
};
