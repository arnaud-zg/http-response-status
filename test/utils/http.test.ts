import { getHttpStatusName, getHttpStatusCatgory } from '../../src/utils/http';

describe('getHttpStatusName', () => {
  it('should return nothing', () => {
    expect(getHttpStatusName(999)).toBeUndefined();
  });

  test.each([
    [100],
    [101],
    [102],
    [200],
    [201],
    [202],
    [203],
    [204],
    [205],
    [206],
    [207],
    [208],
    [226],
    [300],
    [301],
    [302],
    [303],
    [304],
    [305],
    [307],
    [308],
    [400],
    [401],
    [402],
    [405],
    [404],
    [405],
    [406],
    [407],
    [408],
    [409],
    [410],
    [411],
    [412],
    [413],
    [414],
    [415],
    [416],
    [417],
    [418],
    [421],
    [422],
    [423],
    [424],
    [426],
    [428],
    [429],
    [431],
    [444],
    [451],
    [499],
    [500],
    [501],
    [502],
    [503],
    [504],
    [505],
    [506],
    [507],
    [508],
    [510],
    [511],
    [599],
  ])('get name when httpStatusCode is equal to %i', httpStatusCode => {
    expect(getHttpStatusName(httpStatusCode)).toMatchSnapshot();
  });
});

describe('getHttpStatusCatgory', () => {
  it('should return nothing', () => {
    expect(getHttpStatusCatgory(999)).toBeUndefined();
  });

  test.each`
    httpStatusCode | expected
    ${100}         | ${'Informational'}
    ${101}         | ${'Informational'}
    ${102}         | ${'Informational'}
    ${200}         | ${'Success'}
    ${201}         | ${'Success'}
    ${202}         | ${'Success'}
    ${203}         | ${'Success'}
    ${204}         | ${'Success'}
    ${205}         | ${'Success'}
    ${206}         | ${'Success'}
    ${207}         | ${'Success'}
    ${208}         | ${'Success'}
    ${226}         | ${'Success'}
    ${300}         | ${'Redirection'}
    ${301}         | ${'Redirection'}
    ${302}         | ${'Redirection'}
    ${303}         | ${'Redirection'}
    ${304}         | ${'Redirection'}
    ${305}         | ${'Redirection'}
    ${307}         | ${'Redirection'}
    ${308}         | ${'Redirection'}
    ${400}         | ${'ClientError'}
    ${401}         | ${'ClientError'}
    ${402}         | ${'ClientError'}
    ${405}         | ${'ClientError'}
    ${404}         | ${'ClientError'}
    ${405}         | ${'ClientError'}
    ${406}         | ${'ClientError'}
    ${407}         | ${'ClientError'}
    ${408}         | ${'ClientError'}
    ${409}         | ${'ClientError'}
    ${410}         | ${'ClientError'}
    ${411}         | ${'ClientError'}
    ${412}         | ${'ClientError'}
    ${413}         | ${'ClientError'}
    ${414}         | ${'ClientError'}
    ${415}         | ${'ClientError'}
    ${416}         | ${'ClientError'}
    ${417}         | ${'ClientError'}
    ${418}         | ${'ClientError'}
    ${421}         | ${'ClientError'}
    ${422}         | ${'ClientError'}
    ${423}         | ${'ClientError'}
    ${424}         | ${'ClientError'}
    ${426}         | ${'ClientError'}
    ${428}         | ${'ClientError'}
    ${429}         | ${'ClientError'}
    ${431}         | ${'ClientError'}
    ${444}         | ${'ClientError'}
    ${451}         | ${'ClientError'}
    ${499}         | ${'ClientError'}
    ${500}         | ${'ServerError'}
    ${501}         | ${'ServerError'}
    ${502}         | ${'ServerError'}
    ${503}         | ${'ServerError'}
    ${504}         | ${'ServerError'}
    ${505}         | ${'ServerError'}
    ${506}         | ${'ServerError'}
    ${507}         | ${'ServerError'}
    ${508}         | ${'ServerError'}
    ${510}         | ${'ServerError'}
    ${511}         | ${'ServerError'}
    ${599}         | ${'ServerError'}
  `(
    'returns $expected when httpStatusCode is equal to $httpStatusCode',
    ({ httpStatusCode, expected }) => {
      expect(getHttpStatusCatgory(httpStatusCode)).toEqual(expected);
    }
  );
});
