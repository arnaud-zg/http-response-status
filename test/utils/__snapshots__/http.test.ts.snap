// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getHttpStatusName get name when httpStatusCode is equal to 100 1`] = `"CONTINUE"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 101 1`] = `"SWITCHING_PROTOCOLS"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 102 1`] = `"PROCESSING"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 200 1`] = `"OK"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 201 1`] = `"CREATED"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 202 1`] = `"ACCEPTED"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 203 1`] = `"NON_AUTHORITATIVE_INFORMATION"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 204 1`] = `"NO_CONTENT"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 205 1`] = `"RESET_CONTENT"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 206 1`] = `"PARTIAL_CONTENT"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 207 1`] = `"MULTI_STATUS"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 208 1`] = `"ALREADY_REPORTED"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 226 1`] = `"IM_USED"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 300 1`] = `"MULTIPLE_CHOICES"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 301 1`] = `"MOVED_PERMANENTLY"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 302 1`] = `"FOUND"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 303 1`] = `"SEE_OTHER"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 304 1`] = `"NOT_MODIFIED"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 305 1`] = `"USE_PROXY"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 307 1`] = `"TEMPORARY_REDIRECT"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 308 1`] = `"PERMANENT_REDIRECT"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 400 1`] = `"BAD_REQUEST"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 401 1`] = `"UNAUTHORIZED"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 402 1`] = `"PAYMENT_REQUIRED"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 404 1`] = `"NOT_FOUND"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 405 1`] = `"METHOD_NOT_ALLOWED"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 405 2`] = `"METHOD_NOT_ALLOWED"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 406 1`] = `"NOT_ACCEPTABLE"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 407 1`] = `"PROXY_AUTHENTICATION_REQUIRED"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 408 1`] = `"REQUEST_TIMEOUT"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 409 1`] = `"CONFLICT"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 410 1`] = `"GONE"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 411 1`] = `"LENGTH_REQUIRED"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 412 1`] = `"PRECONDITION_FAILED"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 413 1`] = `"PAYLOAD_TOO_LARGE"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 414 1`] = `"REQUEST_URI_TOO_LONG"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 415 1`] = `"UNSUPPORTED_MEDIA_TYPE"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 416 1`] = `"REQUESTED_RANGE_NOT_SATISFIABLE"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 417 1`] = `"EXPECTATION_FAILED"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 418 1`] = `"I_M_A_TEAPOT"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 421 1`] = `"MISDIRECTED_REQUEST"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 422 1`] = `"UNPROCESSABLE_ENTITY"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 423 1`] = `"LOCKED"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 424 1`] = `"FAILED_DEPENDENCY"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 426 1`] = `"UPGRADE_REQUIRED"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 428 1`] = `"PRECONDITION_REQUIRED"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 429 1`] = `"TOO_MANY_REQUESTS"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 431 1`] = `"REQUEST_HEADER_FIELDS_TOO_LARGE"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 444 1`] = `"CONNECTION_CLOSED_WITHOUT_RESPONSE"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 451 1`] = `"UNAVAILABLE_FOR_LEGAL_REASONS"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 499 1`] = `"CLIENT_CLOSED_REQUEST"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 500 1`] = `"INTERNAL_SERVER_ERROR"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 501 1`] = `"NOT_IMPLEMENTED"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 502 1`] = `"BAD_GATEWAY"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 503 1`] = `"SERVICE_UNAVAILABLE"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 504 1`] = `"GATEWAY_TIMEOUT"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 505 1`] = `"HTTP_VERSION_NOT_SUPPORTED"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 506 1`] = `"VARIANT_ALSO_NEGOTIATES"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 507 1`] = `"INSUFFICIENT_STORAGE"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 508 1`] = `"LOOP_DETECTED"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 510 1`] = `"NOT_EXTENDED"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 511 1`] = `"NETWORK_AUTHENTICATION_REQUIRED"`;

exports[`getHttpStatusName get name when httpStatusCode is equal to 599 1`] = `"NETWORK_CONNECT_TIMEOUT_ERROR"`;
